node/src/accountant/db_access_objects/sent_payable_dao.rs

```diff
@@ -292,29 +319,19 @@ mod tests {
         let wrapped_conn = DbInitializerReal::default()
             .initialize(&home_dir, DbInitializationConfig::test_default())
             .unwrap();
-        let tx1 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(1))
-            .status(TxStatus::Pending)
-            .build();
+        let tx1 = TxBuilder::default().hash(make_tx_hash(1)).build();
         let tx2 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(2))
-            .status(TxStatus::Failed)
-            .build();
-        let tx3 = TxBuilder::default()
-            .hash(H256::from_low_u64_le(3))
-            .status(TxStatus::Succeeded(TransactionBlock {
-                block_hash: Default::default(),
-                block_number: Default::default(),
-            }))
+            .hash(make_tx_hash(2))
+            .block(Default::default())
             .build();
         let subject = SentPayableDaoReal::new(wrapped_conn);
-        let txs = vec![tx1, tx2, tx3];
+        let txs = vec![tx1, tx2];
 
         let result = subject.insert_new_records(&txs);
```

Comment:
We were discussing that maybe the code should actually panic if an already confirmed transaction (having the block field populated) is supplied.

Reply:
I think it would be too restrictive, maybe we want to insert confirmed transactions too.